importlib_resources/readers.py

```diff
@@ -83,12 +83,28 @@ def is_file(self):
         return False
 
     def joinpath(self, *descendants):
-        try:
-            return super().joinpath(*descendants)
-        except abc.TraversalError:
-            # One of the paths did not resolve (a directory does not exist).
-            # Just return something that will not exist.
-            return self._paths[0].joinpath(*descendants)
+        # first try to find child in current paths
+        paths = []
```

Comment:
The use of the term 'path' is really overloaded in this method. Let's try to come up with some names that are more indicative of _which_ path/paths are represented.

Reply:
Yes, that's true. My first thought is to rename this to 'candidates', since we are exploring all of our base paths for the same sub path. Another synonym might be better though.